### PR TITLE
Add create wallet/token to wallet list scene search and allow selection of parent wallet for new tokens

### PR DESCRIPTION
**Pull request by peachbits**
---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1202429146875994
  - https://app.asana.com/0/0/1202887459567794

### CODE REVIEW

**Review by paullinator — CHANGES_REQUESTED** [review]

File: src/components/themed/WalletList.tsx [hunk 1]
```diff
@@ -88,6 +91,7 @@ export function WalletList(props: Props) {
 
   // Filter the common wallet list:
   const filteredWalletList = React.useMemo(() => {
+    if (allowedWalletIds != null) return sortedWalletList.filter(item => item.tokenId == null && allowedWalletIds.includes(item.walletId))
```
[the remark this answers is not in the document]
Comment (peachbits) [reply src/components/themed/WalletList.tsx:95]: sortedWalletList has both token and main chain rows so I needed to just get the rows with the main chain, however, I didn't actually use allowedWalletIds and went with a different solution for the parent wallet modal so this needs to get removed altogether.

File: src/components/modals/WalletListModal.tsx [hunk 2]
```diff
@@ -42,6 +42,7 @@ type Props = {
   // Filtering:
   allowedAssets?: EdgeTokenId[]
   excludeAssets?: EdgeTokenId[]
+  allowedWalletIds?: string[]
```
[the remark this answers is not in the document]
Comment (peachbits) [reply src/components/modals/WalletListModal.tsx:45]: Will be deleted as it wasn't actually used

File: src/components/themed/WalletListSwipeable.tsx [hunk 3]
```diff
@@ -42,6 +45,25 @@ export function WalletListSwipeable(props: Props) {
...
+  const createWalletList = React.useMemo(
+    () => (searching ? getCreateWalletList(account, { filteredWalletList: sortedWalletList }) : []),
+    [account, searching, sortedWalletList]
+  )
+
+  const searchedCreateWalletList = React.useMemo(() => {
+    const out: WalletCreateItem[] = []
+    const searchTarget = normalizeForSearch(searchText)
+    for (const item of createWalletList) {
+      const { currencyCode, displayName } = item
```
[the remark this answers is not in the document]
Comment (peachbits) [reply src/components/themed/WalletListSwipeable.tsx:64]: Create wallet rows don't have names. Instead, I added the ability to search the pluginId as well and broke it out into a helper function (this was copy/pasted from another scene) in an additional commit

**Review by paullinator — APPROVED** [review]

**Pull request merged**